### ISSUE TITLE
Crash caused by TimestampColumnReader

### ISSUE THREAD

**Issue by zhli1142015** [oap-project/velox#428]
### Bug description

Crash:

```

#

# A fatal error has been detected by the Java Runtime Environment:

#

#  SIGSEGV (0xb) at pc=0x00007fbf5a802547, pid=2230, tid=0x00007fbf21bd3700

#

# JRE version: OpenJDK Runtime Environment (8.0_382-b05) (build 1.8.0_382-b05)

# Java VM: OpenJDK 64-Bit Server VM (25.382-b05 mixed mode linux-amd64 compressed oops)

# Problematic frame:

# C  [libvelox.so+0x3667547]  void facebook::velox::parquet::PageReader::readWithVisitor<facebook::velox::dwio::common::ColumnVisitor<__int128, facebook::velox::common::AlwaysTrue, facebook::velox::dwio::common::ExtractToReader<facebook::velox::dwio::common::SelectiveIntegerColumnReader>, true> >(facebook::velox::dwio::common::ColumnVisitor<__int128, facebook::velox::common::AlwaysTrue, facebook::velox::dwio::common::ExtractToReader<facebook::velox::dwio::common::SelectiveIntegerColumnReader>, true>&)+0x487

#

# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

#

# An error report file with more information is saved as:

# /home/vsts/work/1/s/core/hs_err_pid2230.log

#

# If you would like to submit a bug report, please visit:

#   https://github.com/adoptium/adoptium-support/issues

# The crash happened outside the Java Virtual Machine in native code.

# See problematic frame for where to report the bug.

#

Exception in thread "Thread-23" java.io.EOFException

  at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:3111)

  at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1620)

  at java.io.ObjectInputStream.readObject(ObjectInputStream.java:503)

  at java.io.ObjectInputStream.readObject(ObjectInputStream.java:461)

  at org.scalatest.tools.Framework$ScalaTestRunner$Skeleton$1$React.react(Framework.scala:839)

  at org.scalatest.tools.Framework$ScalaTestRunner$Skeleton$1.run(Framework.scala:828)

  at java.lang.Thread.run(Thread.java:750)

```

Repro:
`spark.read.format("parquet").load("00000000000000000003.checkpoint.parquet").collect()`
File is uploaded to https://github.com/zhli1142015/velox/blob/check-point-parquet/00000000000000000003.checkpoint.parquet.

By our investigation, this is caused by below override is missed in `TimestampColumnReader`, adding it should fix the issue.
```
  bool hasBulkPath() const override {
    return false;
  }
```

 

### System information

Velox System Info v0.0.2
Commit: 6c9215643d4fca6f17eca1102e8f4bcc99e92465
CMake Version: 3.16.3
System: Linux-5.10.102.1-microsoft-standard-WSL2
Arch: x86_64
C++ Compiler: /usr/bin/c++
C++ Compiler Version: 9.4.0
C Compiler: /usr/bin/cc
C Compiler Version: 9.4.0
CMake Prefix Path: /usr/local;/usr;/;/usr;/usr/local;/usr/X11R6;/usr/pkg;/opt

**Comment by zhli1142015** [comment]
cc @rui-mo , as `TimestampColumnReader` is only supported in OAP velox.

**Comment by rui-mo** [comment]
Thanks! Do you want to open this PR into OAP Velox, or let us fix it in https://github.com/oap-project/velox/commit/b9f081a0fb2102992a58a1588ae0ee596e33530e?

**Comment by zhli1142015** [comment]
I think no need to introduce a separate PR, please help to fix it in the existing commit. Thanks.

**Comment by rui-mo** [comment]
Sure. Can we use the Parquet you provided in a unit test of https://github.com/facebookincubator/velox/pull/4680?

**Comment by zhli1142015** [comment]
ok to me, thanks.

**Comment by rui-mo** [comment]
Covered by https://github.com/oap-project/velox/commit/ead1edec90be38dfedf985eae2216fb035ae8d07.
